common/client-libs/validator-client/src/lib.rs

```diff
@@ -30,6 +31,15 @@ impl Config {
         rest_servers_available_base_urls: Vec<String>,
         mixnet_contract_address: S,
     ) -> Self {
+        // URLs should be in the form: http://127.0.0.1:1317
+        if rest_servers_available_base_urls
+            .iter()
+            .map(|url| url.split(':').count())
```

Comment:
If you're checking for the validity of the URL, it will fail in some cases, for example in ipv6 where the valid address contains multiple `:` characters

Reply:
Yep, I forgot about the ipv6. Better leave the parsing to specialized crates and packages.